src/utils/find.ts

```diff
@@ -28,26 +33,55 @@ function matches(node: VNode, selector: FindAllComponentsSelector): boolean {
   return false
 }
 
+/**
+ * Filters out the null, undefined and primitive values,
+ * to only keep VNode and VNodeArrayChildren values
+ * @param value
+ */
+function nodesAsObject<Node>(
```

Comment:
Is this actually useful? Sees much more complex - what is the benefit of this function? Isn't this effectively `any`?

Reply:
It is different from `any`.

It is used in a `filter` in `aggregateChildren` to remove the potential null objects in the children array, and also removes the primitive values which aren't interesting. 

It is also a type guard, so TS knows that when this function is used in the filter, the array contains only non null values, and non primitive values after filtering.  To my knowledged, this is still too hard for TS to infer these kinds of things by itself, and I believe it's still the recommended way to do it.